tests/OpenTelemetry/Integration/InteroperabilityTest.php

```diff
@@ -1149,4 +1150,60 @@ public function testSpanLinksInteroperabilityAddDuplicates()
             $this->assertNotSame($otelSpanLinks[1], $otelSpanLinks[3]);
         });
     }
+
+    public function testSpanEventsInteroperabilityFromDatadogSpan()
+    {
+        $traces = $this->isolateTracer(function () {
+            $span = start_span();
+            $span->name = "dd.span";
+
+            $spanEvent = new SpanEvent();
+            $spanEvent->name = "event-name";
+            $spanEvent->timestamp = 1720037568765201300;
+            $spanEvent->attributes = [
+                'arg1' => 'value1',
+            ];
+            $span->events[] = $spanEvent;
+
+            /** @var en $OTelSpan */
+            $OTelSpan = Span::getCurrent();
+            $OTelSpanEvent = $OTelSpan->toSpanData()->getEvents()[0];
+
+            // Check the first event
+            $this->assertSame('event-name', $OTelSpanEvent->getName());
+
+            close_span();
+        });
+
+        $this->assertCount(1, $traces[0]);
+        $this->assertSame("[{\"name\":\"event-name\",\"time_unix_nano\":1720037568765201300,\"attributes\":{\"arg1\":\"value1\"}}]", $traces[0][0]['meta']['events']);
```

Comment:
One note here: If an event has an _array_ as one of the attributes, we aren't exploding it out into indices. For example, if there is an attribute `'stringArr' => ['value1', 'value2']` then we should expect the JSON to look like `\"stringArr\":[\"value1\",\"value2\"]`.

Can you make sure this test case gets added to the unit tests?